src/frontend/packages/git/src/shared/scm/scm.service.ts

```diff
@@ -17,10 +17,10 @@ export class GitSCMService {
   ) {
   }
 
-  public getSCM(type: GitSCMType, endpointGuid: string): GitSCM {
+  public getSCM(type: GitSCMType, endpointGuid: string, access_token?: string): GitSCM {
     switch (type) {
       case 'github':
-        return new GitHubSCM(this.gitHubURL, endpointGuid);
+        return (access_token && access_token != "") ? new GitHubSCM(this.gitHubURL, endpointGuid, access_token) : new GitHubSCM(this.gitHubURL, endpointGuid);
```

Comment:
you can move the accessToken validation in the constructor of GitHubSCM. -> the constructor decides if the given access code exists, is correct, has something missing.

```suggestion
        return new GitHubSCM(this.gitHubURL, endpointGuid, accessToken)
```

Reply:
Done 👍🏻